### PR TITLE
Don't generate scripts if they will not be added to the user project

### PR DESCRIPTION
**Pull request by amorde**
Follow up to #7181 - if we aren't going to add the scripts to the user project, don't generate the files at all and don't write them to disk.

cc @keith

### CODE REVIEW

File: lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb [hunk 1]
```diff
@@ -6,6 +6,11 @@ class PodsProjectGenerator
         # project and the relative support files.
         #
         class AggregateTargetInstaller < TargetInstaller
+          # @return [AggregateTarget] target
+          #         the target to install
+          #
+          attr_reader :target
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb]: why is this needed? Isnt it available from the base class?

Comment (amorde) [reply lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb]: was trying to update the type annotation to be `AggregateTarget` for this subclass - is there an easier way to do that?

File: lib/cocoapods/target/aggregate_target.rb [hunk 2]
```diff
@@ -191,6 +191,19 @@ def uses_swift?
...
+    # @return [Boolean] Whether the target contains any resources
+    #
+    def has_resources?
+      !resource_paths_by_config.values.all?(&:empty?)
+    end
+
+    # @return [Boolean] Whether the target contains framework to be embedded into
+    #         the user target
+    #
+    def has_frameworks?
```
Comment (amorde) [inline lib/cocoapods/target/aggregate_target.rb]: rubocop doesn't like the name of these - any suggestions

File: spec/unit/installer/user_project_integrator/target_integrator_spec.rb [hunk 3]
```diff
@@ -49,16 +49,6 @@ module Pod
...
 
-        it 'fixes the copy resource scripts of legacy installations' do
-          @target_integrator.integrate!
-          target = @target_integrator.send(:native_targets).first
-          phase_name = @copy_pods_resources_phase_name
-          phase = target.shell_script_build_phases.find { |bp| bp.name == phase_name }
-          phase.shell_script = %("${PODS_ROOT}/Pods-resources.sh"\n)
-          @target_integrator.integrate!
-          phase.shell_script.strip.should == '"${PODS_ROOT}/Target Support Files/Pods/Pods-resources.sh"'
-        end
```
Comment (amorde) [inline spec/unit/installer/user_project_integrator/target_integrator_spec.rb]: The "legacy installations" here refers to pre-0.34, I think we can drop any special handling for that by now

File: lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb (hunk 1 above)
Comment (segiddins) [reply lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb]: You could use yard's `@!parse` macro: https://www.rubydoc.info/gems/yard/file/docs/Tags.md#parse

File: lib/cocoapods/target/aggregate_target.rb (hunk 2 above)
Comment (segiddins) [reply lib/cocoapods/target/aggregate_target.rb]: `includes_...?`

**Comment by dnkoutso** [comment]
Does this change mean we actually *do* create the files on disk (the `.sh` scripts) but we just don't add the script phase if there is nothing to copy or embed?

**Comment by amorde** [comment]
> Does this change mean we actually do create the files on disk (the .sh scripts) but we just don't add the script phase if there is nothing to copy or embed?

yep exactly that

**Comment by dnkoutso** [comment]
OK. Was wondering if we wanted to take it further and not even write the `.sh` files! but this change is great as it is.

**Comment by amorde** [comment]
Err sorry, this PR is actually doing that @dnkoutso - now we won't make the files at all if they aren't going to be added to the build phases :)

**Comment by amorde** [comment]
what I meant is right now we don't add them to the build phase if there aren't resources / frameworks, but on master we still make the actual files.
